sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -181,7 +181,7 @@ public TableClient(string connectionString, string tableName, TableClientOptions
             _isCosmosEndpoint = TableServiceClient.IsPremiumEndpoint(connString.TableStorageUri.PrimaryUri);
             var perCallPolicies = _isCosmosEndpoint ? new[] { new CosmosPatchTransformPolicy() } : Array.Empty<HttpPipelinePolicy>();
 
-            options ??= new TableClientOptions();
+            options ??= new TablesClientOptions();
```

Comment:
Are we concerned with the options being mutated after construction?   Should we consider making a defensive copy?

Reply:
I know we do this in EventHubs, but is that a pattern followed by all clients?

Reply:
I know @pakrym was working on a solution to this problem - I'm not sure if it is ready yet.

Reply:
I don't know;  it may not be as much of an issue for HTTP-based clients, since there's a distinct request/response and changing the options between requests could be desirable?

Reply:
`HttpPipelineBuilder.Build` effectively copies most of the settings.